packages/webamp-modern-2/src/skin/AudioPlayer.ts

```diff
@@ -131,10 +157,48 @@ export class AudioPlayer {
     return this._audio.currentTime / this._audio.duration;
   }
 
+  getState(): number {
+    if(!!this.__isStop) { // To distinc with pause
+      return STATUS_STOPPED; 
+    }
+    const audio = this._audio;
+    if(!audio.ended && !audio.paused) {
+      return STATUS_PLAYING
+    } 
+    else
+    if(audio.ended){
+      return STATUS_STOPPED
+    } 
+    else 
+    if(audio.paused){
+      return STATUS_PAUSED
+    } 
+  }
+
+  // doStateChanged() {
+  //   this.trigger('play')
+  // }
+
+  // onStateChange(cb: () => void): () => void {
+  //   const handler = () => {
+  //     // console.log('audio.onPlay!')
+  //     cb();
+  //   }
+  //   this._audio.addEventListener("playing", handler);
+  //   this._audio.addEventListener("pause", handler);
+  //   this._audio.addEventListener("ended", handler);
+  //   const dispose = () => {
+  //     this._audio.removeEventListener("playing", handler);
+  //     this._audio.removeEventListener("pause", handler);
+  //     this._audio.removeEventListener("ended", handler);
+  //   };
+  //   return dispose;
+  // }
+
```

Comment:
In general, lets remove commented code, or -- if you think it's worth keeping for some reason -- explain that reason in a comment.
```suggestion
```